src/components/Link.tsx

```diff
@@ -24,66 +23,62 @@ import {
 } from '#/lib/strings/url-helpers'
 import {useModalControls} from '#/state/modals'
 import {router} from '#/routes'
+import {Text} from '#/components/Typography'
 
-export type LinkProps = Omit<
-  ButtonProps,
-  'style' | 'onPress' | 'disabled' | 'label'
+/**
+ * Only available within a `Link`, since that inherits from `Button`.
+ * `InlineLink` provides no context.
+ */
+export {useButtonContext as useLinkContext} from '#/components/Button'
+
+type BaseLinkProps = Pick<
+  Parameters<typeof useLinkProps<AllNavigatorParams>>[0],
+  'to'
 > & {
-  /**
-   * `TextStyle` to apply to the anchor element itself. Does not apply to any children.
-   */
-  style?: StyleProp<TextStyle>
   /**
    * The React Navigation `StackAction` to perform when the link is pressed.
    */
   action?: 'push' | 'replace' | 'navigate'
+
   /**
-   * If true, will warn the user if the link text does not match the href. Only
-   * works for Links with children that are strings i.e. text links.
+   * If true, will warn the user if the link text does not match the href.
+   *
+   * Note: atm this only works for `InlineLink`s with a string child.
    */
   warnOnMismatchingTextChild?: boolean
-  label?: ButtonProps['label']
-} & Pick<Parameters<typeof useLinkProps<AllNavigatorParams>>[0], 'to'>
+}
 
-/**
- * A interactive element that renders as a `<a>` tag on the web. On mobile it
- * will translate the `href` to navigator screens and params and dispatch a
- * navigation action.
- *
- * Intended to behave as a web anchor tag. For more complex routing, use a
- * `Button`.
- */
-export function Link({
-  children,
+export function useLink({
```

Comment:
Did some work in this file to differentiate between inline links and other custom links. This hook powers both, and can be used elsewhere for the same effect.